app/models/user.rb

```diff
@@ -16,6 +16,14 @@ class User < ApplicationRecord
 
   mount_uploader :image, ImageUploader
 
+  def self.all_liking_post(post_id)
+    join(:likes).where(likes: { post_id: params[:id] } )
+  end
+
+  def post_like(post_id)
+    likes.where(post_id: post_id).first
```

Comment:
How do we make sure that the like only belongs to the current user? When I ran the query it gave me back the first like on the specific post that belongs to a different user.
I was wondering if I can do something like this maybe?
`def post_like(post_id)
    Like.find_by(post_id: post_id, user_id: self)
  end`

Reply:
@muazzamnashat you would run `current_user.post_like(post_id)`
See line 5 in `followers_controller.rb`

Reply:
@muazzamnashat 
```
# get all the likes that belong to the current user
current_user.likes

# Find all the likes that matches the current user and with post_id matching post_id
current_user.likes.where(post_id: post_id)

# Find first like that matches the current user and with post_id matching post_id
current_user.likes.where(post_id: post_id).first
```